### PR TITLE
OpenSSL => 3.3.1 + ruby update + ncurses update + python update

### LINKED ISSUE #9923: 🐞 Bug - Installation issue - OPENSSL_3.2.0 not found
I tried to install chromebrew and the process stopped at this point:
OPENSSL_3.2.0 not found
Screenshot 2024-06-07 19 34 53

I hope there's a way to fix it or any workaround is appreciated. Thank you

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #9923

- Adds `ruby` built against this OpenSSL.
- Adds updated `ncurses`.
- Adds updated `python3`.

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=openssl crew update ; yes | crew upgrade
```

### CODE REVIEW

File: packages/python3.rb [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
-  version '3.12.3'
+  version '3.12.4'
   license 'PSF-2.0'
   compatibility 'all'
   source_url "https://www.python.org/ftp/python/#{version}/Python-#{version}.tar.xz"
-  source_sha256 '56bfef1fdfc1221ce6720e43a661e3eb41785dd914ce99698d8c7896af4bdaa1'
+  source_sha256 'f6d419a6d8743ab26700801b4908d26d97e8b986e14f95de31b32de2b0e79554'
   binary_compression 'tar.zst'
 
   binary_sha256({
```
Comment (uberhacker) [inline packages/python3.rb:13]: We're missing binaries here?

Comment (uberhacker) [reply packages/python3.rb:13]: Also noticed filelist updates were missing too.

Comment (satmandu) [reply packages/python3.rb:13]: Python is still building on my i686 builder.

Once that is done I will upload binaries and file lists...

Comment (satmandu) [reply packages/python3.rb:13]: And only the ln will I take this PR out of draft. 😅

Comment (uberhacker) [reply packages/python3.rb:13]: Gotcha

Comment (satmandu) [reply packages/python3.rb:13]: Done.

**Review by uberhacker — APPROVED** [review]
Hopefully, this fixes #9923.

**Pull request merged**